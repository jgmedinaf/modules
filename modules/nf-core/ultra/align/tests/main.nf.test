nextflow_process {

    name "Test Process ULTRA_ALIGN"
    config "./nextflow.config"
    script "../main.nf"
    process "ULTRA_ALIGN"

    tag "modules"
    tag "modules_nfcore"
    tag "ultra"
    tag "ultra/align"
    tag "ultra/index"
    tag "gunzip"
    tag "gnu"
    tag "gnu/sort"

    test("test_ultra_align") {

        setup {
            run("GUNZIP") {
                script "../../../gunzip/main.nf"

                process {
                    """
                    input[0] = [
                        [id:'test',single_end:false],
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/pacbio/fastq/test_hifi.fastq.gz',checkIfExists:true)
                        ]
                    """
                }
            }

            run("GNU_SORT") {
                script "../../../gnu/sort/main.nf"
                process {
                    """
                    input[0] = [
                        [id:'test',single_end:false],
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf',checkIfExists:true)
                        ]
                    """
                }
            }
            
            run("ULTRA_INDEX") {
                script "../../../ultra/index/main.nf"
                process {
                    """
                    input [0] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta',checkIfExists:true)
                    input [1] = GNU_SORT.out.sorted.map{it[1]}
                    """
                }
            }
        }
        when {
            params {
                gnu_sort_args = '-k1,1 -k4,4n'
                ultra_index_args = '--disable_infer'
            }
            process {
                """
                input[0] = GUNZIP.out.gunzip
                input[1] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta',checkIfExists:true)
                input[2] = ULTRA_INDEX.out.index
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    bam(process.out.bam[0][1]).getReadsMD5()
                ).match() }
            )
        }
    }
}