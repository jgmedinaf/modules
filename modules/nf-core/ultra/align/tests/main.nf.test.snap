{
    "test_ultra_align": {
        "content": [
            [
                "versions.yml:md5,49ee786f03311445a52ddc0e4136de60"
            ],
            "4a12eec345cffd63b51bd3d29d89f13c"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.1"
        },
        "timestamp": "2025-05-14T15:07:29.874106796"
    }
}